src/components/SidebarContent.vue

```diff
@@ -188,18 +188,27 @@ components: { SearchFilters, DatasetCard, ContextCard },
       this.searchSciCrunch(this.searchInput);
     },
     searchSciCrunch: function (search, filter=undefined) {
+      this.lastSearchInput = search;
       this.loadingCards = true;
       this.results = [];
       this.disableCards();
       let params = this.createParams(filter, this.start, this.numberPerPage)
       this.callSciCrunch(this.apiLocation, this.searchEndpoint, search, params).then((result) => {
-        this.sciCrunchError = false
-        this.resultsProcessing(result)
-        this.$refs.content.style['overflow-y'] = 'scroll'
+        //Only process if the search term is the same as the last search term.
+        //This avoid old search being displayed.
+        if (this.lastSearchInput == search) {
```

Comment:
I think facets also use this method to update the results. So we probably need a check for them otherwise facets won't trigger a search

Reply:
As long as the last entered search term is the same as the one pass into this method, the result will be updated. Changing the facet does not change the search term so it should be fine.